src/service/viewer-impl.js

```diff
@@ -257,6 +254,13 @@ export class Viewer {
     /** @const @private {!Promise<boolean>} */
     this.isTrustedViewer_ = trustedViewerPromise;
 
+    const messagingOriginPromise = new Promise(resolve => {
+      /** @const @private {!function(string)} */
+      this.messagingOriginResolver_ = resolve;
+    });
+    /** @const @private {!Promise<string>} */
+    this.messagingOrigin_ = messagingOriginPromise;
```

Comment:
We also need tests for this in `test-viewer.js`. It should be very similar to test-set for `getReferrerUrl`.